### Arquivos disponíveis em <https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/microdados/enem>

In [58]:
import pandas as pd

In [59]:
url_or_file = 'MICRODADOS_ENEM_2019.csv'

In [60]:
colunas = ['TP_PRESENCA_CN', 
           'TP_PRESENCA_CH', 
           'TP_PRESENCA_LC', 
           'TP_PRESENCA_MT', 
           'NU_NOTA_CN', 
           'NU_NOTA_CH',
           'NU_NOTA_LC',
           'NU_NOTA_MT',
           'NU_NOTA_REDACAO']

In [61]:
dados = pd.read_csv(url_or_file, usecols = colunas, sep = ';', encoding = 'ISO-8859-1')

In [62]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5095171 entries, 0 to 5095170
Data columns (total 9 columns):
 #   Column           Dtype  
---  ------           -----  
 0   TP_PRESENCA_CN   int64  
 1   TP_PRESENCA_CH   int64  
 2   TP_PRESENCA_LC   int64  
 3   TP_PRESENCA_MT   int64  
 4   NU_NOTA_CN       float64
 5   NU_NOTA_CH       float64
 6   NU_NOTA_LC       float64
 7   NU_NOTA_MT       float64
 8   NU_NOTA_REDACAO  float64
dtypes: float64(5), int64(4)
memory usage: 349.9 MB


In [63]:
dados.head(10)

TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               0               0               0               0         NaN   
1               0               0               0               0         NaN   
2               0               0               0               0         NaN   
3               1               1               1               1       483.8   
4               1               1               1               1       513.6   
5               1               1               1               1       563.7   
6               1               1               1               1       484.6   
7               0               0               0               0         NaN   
8               0               0               0               0         NaN   
9               1               1               1               1       543.9   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO  
0         NaN         NaN         NaN              NaN  
1         NaN         NaN         NaN              NaN  
2         NaN         NaN         NaN              NaN  
3       503.6       537.3       392.0            460.0  
4       575.5       570.7       677.0            860.0  
5       644.9       564.2       675.3            800.0  
6       488.4       507.2       594.7            600.0  
7         NaN         NaN         NaN              NaN  
8         NaN         NaN         NaN              NaN  
9       548.1       502.5       480.7            400.0

In [64]:
colunas_presenca = ['NU_NOTA_CN', 'NU_NOTA_CH', 'NU_NOTA_LC', 'NU_NOTA_MT', 'NU_NOTA_REDACAO']
dados_presentes = dados[colunas_presenca]

In [65]:
dados_presentes.head(10)

NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO
0         NaN         NaN         NaN         NaN              NaN
1         NaN         NaN         NaN         NaN              NaN
2         NaN         NaN         NaN         NaN              NaN
3       483.8       503.6       537.3       392.0            460.0
4       513.6       575.5       570.7       677.0            860.0
5       563.7       644.9       564.2       675.3            800.0
6       484.6       488.4       507.2       594.7            600.0
7         NaN         NaN         NaN         NaN              NaN
8         NaN         NaN         NaN         NaN              NaN
9       543.9       548.1       502.5       480.7            400.0

In [66]:
dados_presentes = dados_presentes.dropna(how = 'all', axis = 0)

In [67]:
dados_presentes.head(10)

NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO
3        483.8       503.6       537.3       392.0            460.0
4        513.6       575.5       570.7       677.0            860.0
5        563.7       644.9       564.2       675.3            800.0
6        484.6       488.4       507.2       594.7            600.0
9        543.9       548.1       502.5       480.7            400.0
11       518.2       351.7       417.6       459.0            500.0
12       373.8       393.6       451.0       484.7            300.0
13       458.9       441.6       413.0       399.0            240.0
16       390.1       453.0       515.2       461.5            520.0
18       415.7       379.2       470.2       375.1            560.0

In [68]:
print(f'Total de inscritos no ENEM 2019: {(len(dados)):,.0f}.')
print(f'Total de inscritos que registraram alguma presença: {(len(dados_presentes)):,.0f} ({(len(dados_presentes) / len(dados)):.2%}).')
print(f'Total de inscritos ausentes nos dois dias de provas: {(len(dados) - len(dados_presentes)):,.0f} ({((len(dados) - len(dados_presentes)) / len(dados)):.2%}).')

Total de inscritos no ENEM 2019: 5,095,171.
Total de inscritos que registraram alguma presença: 3,931,471 (77.16%).
Total de inscritos ausentes nos dois dias de provas: 1,163,700 (22.84%).


Informações importantes:

Foram excluídos da base do microdados os registros dos participantes que realizaram as provas: 543,
544, 545 e 546 de Ciências da Natureza; 547, 548, 549, 550 e 564 de Ciências Humanas; 551, 552,
553, 554 e 565 de Linguagens e Códigos; e 555, 556, 557 e 558 de Matemática. Isso porque o
número de participantes que realizaram essas provas era muito pequeno, o que permitiria a sua
identificação indevida (LGPD).

Fonte: Leia_Me_Enem_2019.pdf (p. 7)
Disponível em <https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/microdados/enem>, acesso em 28 ago. 2022.

<https://vestibular.brasilescola.uol.com.br/enem/enem-2019-registra-menor-abstencao-historia/346656.html>
<https://g1.globo.com/educacao/enem/2019/noticia/2019/11/10/enem-2019-tem-menor-taxa-de-abstencao-da-historia-diz-ministerio.ghtml>